web/server/vue-cli/src/mixins/analysis-info-handling.mixin.js

```diff
@@ -0,0 +1,268 @@
+                .map(checker =>
+                  this.checkersGroupedAndSorted[analyzer][group][checker][1]
+                    .enabled ? 1 : 0)
+                .reduce((a, b) => a + b, 0),
+              /* [CountKeys.Disabled]: */ -1, // Will be updated later!
+              // [CountKeys.Total]:
+              Object.keys(
+                this.checkersGroupedAndSorted[analyzer][group]
+              ).length
+            ]
+            ]))
+      ]));
+    Object.keys(this.checkerGroupCounts).map(analyzer =>
+      Object.keys(this.checkerGroupCounts[analyzer]).map(group => {
+        this.checkerGroupCounts[analyzer][group][CountKeys.Disabled] =
+          this.checkerGroupCounts[analyzer][group][CountKeys.Total] -
+          this.checkerGroupCounts[analyzer][group][CountKeys.Enabled];
+      }));
+    Object.keys(this.checkerGroupCounts).map(analyzer => {
+      const sum = Object.values(this.checkerGroupCounts[analyzer])
+        .reduce((as, bs) => [
+          as[CountKeys.Enabled]  + bs[CountKeys.Enabled],
+          as[CountKeys.Disabled] + bs[CountKeys.Disabled],
+          as[CountKeys.Total]    + bs[CountKeys.Total]
+        ]);
+      this.checkerGroupCounts[analyzer][GroupKeys.AnalyzerTotal] = sum;
+    });
+
+    return this.checkersGroupedAndSorted;
+  }
+}
+
+const AnalysisInfoHandlingMixin = {
+  mixins: [ VersionMixin ],
+  props: {
+    runId: { type: Object, default: () => null },
+    runHistoryId: { type: Object, default: () => null },
+    reportId: { type: Object, default: () => null },
+  },
+
+  methods: {
+    loadAnalysisInfo() {
+      const analysisInfoFilter = new AnalysisInfoFilter({
+        // Query a run's analysis info only if a run history ID is not explicit.
+        runId: (!this.runHistoryId ? this.runId : null),
+
+        runHistoryId: this.runHistoryId,
+        reportId: this.reportId
+      });
+      const limit = null;
+      const offset = 0;
+
+      return new Promise(resolve => {
+        ccService.getClient()
+          .getAnalysisInfo(analysisInfoFilter, limit, offset,
+            handleThriftError(aiResult => {
+              var analysisInfo = new AnalysisInfo();
+              analysisInfo.cmds = aiResult.map(r => r.analyzerCommand);
+              analysisInfo.checkers = aiResult.map(r => r.checkers)
+                .reduce(mergeReduceCheckerStatuses, {});
+
+              if (Object.keys(analysisInfo.checkers).length) {
+                analysisInfo.checkerInfoAvailability =
+                  CheckerInfoAvailability.Available;
+              } else {
+                analysisInfo.checkerInfoAvailability =
+                  CheckerInfoAvailability.UnknownReason;
+                this.decideNegativeCheckerStatusAvailability(analysisInfo);
+              }
+
+              analysisInfo.populateAnalyzers();
+
+              resolve(analysisInfo);
+            }));
+      });
+    },
+
+    decideNegativeCheckerStatusAvailability(analysisInfo) {
```

Comment:
Same problem here. It would be better to get runId, runHistorId and reportId as a parameter of the function.